packages/react/src/analytics/integrations/GA4/types/index.ts

```diff
@@ -0,0 +1,65 @@
+import type {
+  EventData,
+  IntegrationOptions,
+  LoadIntegrationEventData,
+  SetUserEventData,
+  trackTypes,
+  TrackTypesValues,
+} from '@farfetch/blackout-analytics';
+import type { Schema } from '../../shared/types/shared.types';
+
+export type Schemas = Record<string, Schema>;
+
+export interface GA4IntegrationOptions extends IntegrationOptions {
+  measurementId: string;
+  enableAutomaticPageViews?: boolean;
+  nonInteractionEvents?: NonInteractionEvents;
+  productMappings?: ProductMappings;
+  scopeCommands?: ScopeCommands;
+  schemas?: Schemas;
+  ScopeCommands?: boolean;
+  dataLayerName?: string;
+  loadScript?: () => Promise<void>;
+  onPreProcessCommands?: OnPreProcessCommandsHandler;
+}
+
+export type ProductMappings = Record<string, string | string[]>;
+
+export type NonInteractionEvents = Record<string, boolean>;
+
+export type OnPreProcessCommandsHandler = (
+  commandList: GA4CommandList,
+  data:
+    | EventData<TrackTypesValues>
+    | SetUserEventData
+    | LoadIntegrationEventData,
+) => GA4CommandList;
+
+export type GA4CommandList = GA4Command[];
+export type GA4Command = [command: string, ...args: unknown[]];
+
+export type EventScopeCommandsHandlers = {
+  extras?: (data: EventData<typeof trackTypes.TRACK>) => GA4CommandList;
```

Comment:
In `packages/analytics/src/types/analytics.types.ts` there are the following type aliases that you can use instead:

```js

export type PageviewEventData = EventData<typeof trackTypes.PAGE>;
export type ScreenviewEventData = EventData<typeof trackTypes.SCREEN>;
export type TrackEventData = EventData<typeof trackTypes.TRACK>;
```
Use them to all the types where you have a specific `EventData` type.